xrfeitoria/utils/publish_plugins.py

```diff
@@ -1,98 +1,191 @@
 """Publish plugins to zip files.
 
->>> python -m xrfeitoria.utils.publish_plugins --unreal -s 0.5.0-Unreal5.2-Windows
->>> python -m xrfeitoria.utils.publish_plugins --blender -s 0.5.0-None-None
+>>> python -m xrfeitoria.utils.publish_plugins --help
 """
+import os
+import platform
+import re
+import subprocess
+from contextlib import contextmanager
 from pathlib import Path
-from typing import Literal, Optional
+from typing import List, Optional
 
 from loguru import logger
 
-from .. import __version__
-from ..data_structure.constants import PathLike, plugin_name_blender, plugin_name_unreal
+from ..data_structure.constants import plugin_name_blender, plugin_name_pattern, plugin_name_unreal
+from ..utils import setup_logger
+from ..version import __version__, __version_tuple__
+from .runner import UnrealRPCRunner
 
 root = Path(__file__).parent.resolve()
 project_root = root.parents[1]
+src_root = project_root / 'src'
+dist_root = src_root / 'dist'
+dist_root.mkdir(exist_ok=True, parents=True)
+
+
+@contextmanager
+def working_directory(path):
+    """Changes working directory and returns to previous on exit."""
+    prev_cwd = Path.cwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(prev_cwd)
 
 
 def _make_archive(
-    plugin_folder: PathLike,
-    zip_name: Optional[str] = None,
-    folder_name: Optional[str] = None,
+    src_folder: Path,
+    dst_path: Optional[Path] = None,
+    folder_name_inside_zip: Optional[str] = None,
+    filter_names: List[str] = ['.git', '.idea', '.vscode', '.gitignore', '.DS_Store', '__pycache__', 'Intermediate'],
```

Comment:
Using a mutable default value as an argument is not recommended.
Considering one of the following approaches:
* Defaults to None, and `if filter_names is None: filter_names = [...]`
* Defaults to Tuple

Reply:
Accepted in [`9dca9a9` (#15)](https://github.com/openxrlab/xrfeitoria/pull/15/commits/9dca9a9d5c62d5cb66b3c1a5b226f1ee5282aa65)